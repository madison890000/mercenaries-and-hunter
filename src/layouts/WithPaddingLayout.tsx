import { PropsWithChildren } from 'react';

const WithPaddingLayout: React.FC<PropsWithChildren> = ({ children }) => (
  <div
    style={{
      margin: 'var(--base-layout-padding)',
      padding: 'var(--base-padding)',
      background: 'var(--color-white)',
      borderRadius: 'var(--base-border-radius)'
    }}
  >
    {children}
  </div>
);
export default WithPaddingLayout;
