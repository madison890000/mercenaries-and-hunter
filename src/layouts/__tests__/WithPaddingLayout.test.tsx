import React from 'react';
import renderer from 'react-test-renderer';
import WithPaddingLayout from "../WithPaddingLayout";

test('renders WithPaddingLayout', () => {
    const WithPaddingLayoutDom = renderer.create(
        <WithPaddingLayout>
            Test Content
        </WithPaddingLayout>
    ).toJSON();
    expect(WithPaddingLayoutDom).toMatchSnapshot();
});
