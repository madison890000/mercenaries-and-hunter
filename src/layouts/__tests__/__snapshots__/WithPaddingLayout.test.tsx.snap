// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WithPaddingLayout 1`] = `
<div
  style={
    Object {
      "background": "var(--color-white)",
      "borderRadius": "var(--base-border-radius)",
      "margin": "var(--base-layout-padding)",
      "padding": "var(--base-padding)",
    }
  }
>
  Test Content
</div>
`;
